File: src/engine/logmessages/errorlogmessage.cpp
#include "errorlogmessage.h"

#include "diskutil.h"


ErrorLogMessage::ErrorLogMessage(const QString &message) :
    LogMessage(),
    m_message(message) {
}

ErrorLogMessage::~ErrorLogMessage() {
}

void ErrorLogMessage::log() {

    DiskUtil::appendToLogFile("errors", m_message);
}

File: src/logs/log_messages/error_log_message.rs
use super::LogMessage;

#[derive(Debug)]
pub struct ErrorLogMessage {
    message: String,
}

impl ErrorLogMessage {
    pub fn new(message: String) -> Self {
        Self { message }
    }
}

impl LogMessage for ErrorLogMessage {
    fn get_log(&self) -> &str {
        "errors"
    }

    fn get_message(&self) -> &str {
        &self.message
    }
}
